### PR TITLE
Grub EFI images for x86 / x86_64

### PR DESCRIPTION
**Pull request by sduponch**
- Add option in config to build a UEFI Grub image with legacy grub support.
- Update the x86 Makefile to generate an UEFi grub image that append kernel vfat and grub modules partition to a copy of the generated legacy image.
- EFI image generate an UUID GPT signature to replace the too small legacy part signature.
- EFI image use the grub search files module to find the boot partition (as grub does not have GPT uuid part search support).
- Fork grub build with efi platform via a common.mk Makefile include to not fork the whole grub package and patches (host need both of legacy and efi modules to setup the bootloader on the efi image).
- Add gptfdisk tools packages to convert legacy partition table to GPT with a patch to build only gdisk.
- Update platform.sh to support GPT UUID root signature in cmdline (use awk  that is embeded in the pivot root of common.sh to reorder bytes of the partition signatures).
- Grub-efi package declare and enable a kmod-efi-vars kernel module to enable EFI related Kernel config option that are required to boot with a console on HDMI.
- Generate VDI and VMDK EFI images

Known limitations : 
- Sysupgrade -p is forced for EFI/GPT images.

Tested on : 
- Virtualbox legacy mode
- Virtualbox EFI mode
- Intel Nuc DE3815TYKE legacy mode
- Intel Nuc DE3815TYKE EFI mode
- Giada F-210 EFI only bios
- Intel Cherry trail platform with EFI only bios

Not tested on x86 32bits platform

### CODE REVIEW

File: target/linux/x86/image/Makefile [hunk 1]
```diff
@@ -41,6 +44,7 @@ ifneq ($(GRUB_TERMINALS),)
 endif
 
 SIGNATURE:=$(shell dd if=/dev/urandom bs=4 count=1 2>/dev/null | hexdump -v -e '"%08x"')
+EFI_SIGNATURE:=$(strip $(shell uuidgen | sed "s/[a-zA-Z0-9][a-zA-Z0-9][a-zA-Z0-9][a-zA-Z0-9]$$/0002/" | tr '[a-z]' '[A-Z]'))
```
Comment (gregdel) [inline target/linux/x86/image/Makefile]: [a-zA-Z0-9][a-zA-Z0-9][a-zA-Z0-9][a-zA-Z0-9] != [a-zA-Z0-9]{4} ?

File: target/linux/x86/base-files/lib/upgrade/platform.sh [hunk 2]
```diff
@@ -30,6 +30,23 @@ platform_export_bootdevice() {
 		esac
 
 		case "$disk" in
+			PARTUUID=[A-F0-9][A-F0-9][A-F0-9][A-F0-9][A-F0-9][A-F0-9][A-F0-9][A-F0-9]-[A-F0-9][A-F0-9][A-F0-9][A-F0-9]-[A-F0-9][A-F0-9][A-F0-9][A-F0-9]-[A-F0-9][A-F0-9][A-F0-9][A-F0-9]-[A-F0-9][A-F0-9][A-F0-9][A-F0-9][A-F0-9][A-F0-9][A-F0-9][A-F0-9]0002)
```
Comment (gregdel) [inline target/linux/x86/base-files/lib/upgrade/platform.sh]: Ugly

File: package/boot/grub2/grub2-efi/Makefile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+define KernelPackage/efi-vars
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=EFI VARS modules
+  DEPENDS:=@EFI_IMAGES
+  KCONFIG:= \
+	CONFIG_EFI=y \
+	CONFIG_EFI_STUB=n \
+	CONFIG_ACPI_BGRT=y \
+        CONFIG_EFI_FAKE_MEMMAP=y \
```
Comment (gregdel) [inline package/boot/grub2/grub2-efi/Makefile]: Not aligned

File: target/linux/x86/base-files/lib/upgrade/platform.sh [hunk 4]
```diff
@@ -30,6 +30,23 @@ platform_export_bootdevice() {
...
+				uuid="${disk#PARTUUID=}"
+				uuid="${uuid%0002}0002"
+				for disk in $(find /dev -type b); do
+					set -- $(dd if=$disk bs=1 skip=$((2*512+128+16)) count=16 2>/dev/null | hexdump -v -e '4/1 "%02x"' | awk '{ \
+							for(i=1;i<9;i=i+2) first=substr($0,i,1) substr($0,i+1,1) first; \
```
Comment (gregdel) [inline target/linux/x86/base-files/lib/upgrade/platform.sh]: scary !

File: target/linux/x86/image/Makefile (hunk 1 above)
Comment (sduponch) [inline target/linux/x86/image/Makefile]: seb@compile:~/openwrt ‹x86_uefi*›$ uuidgen | sed "s/[a-zA-Z0-9][a-zA-Z0-9][a-zA-Z0-9][a-zA-Z0-9]$/0002/"
72d11a22-997c-41fe-8ffc-762c2b4f0002
seb@compile:~/openwrt ‹x86_uefi*›$ uuidgen | sed "s/[a-zA-Z0-9]{4}$/0002/"                              
7d451cb5-1f19-436a-8ba1-67beca225bc3

{4} is not supported

File: target/linux/x86/base-files/lib/upgrade/platform.sh (hunk 2 above)
Comment (sduponch) [inline target/linux/x86/base-files/lib/upgrade/platform.sh]: same as sed

File: target/linux/x86/base-files/lib/upgrade/platform.sh (hunk 4 above)
Comment (sduponch) [inline target/linux/x86/base-files/lib/upgrade/platform.sh]: Same method as openwrt, i have only updated offset and pipe to an awk to reorder hex bytes

**Pull request closed without merging**